### PR TITLE
docs : add curvefs client design

### PR DESCRIPTION
**Pull request by xu-chaojie**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: docs/cn/curvefs-client-design.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+## 1 概要
+CurveFS的client通过对接fuse，实现完整的文件系统功能。CurveFS client支持数据存储在两种后端，分别是S3兼容的对象存储和Curve块存储中（其他块存储的支持也在计划中），目前已支持S3存储后端，存储到CurveBS后端尚在完善中，后续还可能支持S3和Curve块混合存储，让数据根据冷热程度在S3与Curve块之间流动。
+
+## 2 功能介绍
+### 2.1 提供标准posix文件系统接口
```
Comment (aspirer) [inline docs/cn/curvefs-client-design.md:5]: S3、POSIX、CurveBS等大小写统一起来，建议按我备注的格式

File: docs/cn/curvefs-client-design.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+CurveFS client通过对接libfuse 的 lowlevel fuse api，支持fuse用户态文件系统，实现标准posix文件系统接口。
+
+### 2.2 支持s3存储引擎存储数据
```
Comment (aspirer) [inline docs/cn/curvefs-client-design.md:8]: 重点强调CurveBS后端，S3作为附加特性，先介绍BS

File: docs/cn/curvefs-client-design.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+## 1 概要
+CurveFS的client通过对接fuse，实现完整的文件系统功能。CurveFS client支持数据存储在两种后端，分别是S3兼容的对象存储和Curve块存储中（其他块存储的支持也在计划中），目前已支持S3存储后端，存储到CurveBS后端尚在完善中，后续还可能支持S3和Curve块混合存储，让数据根据冷热程度在S3与Curve块之间流动。
```
Comment (aspirer) [inline docs/cn/curvefs-client-design.md:2]: BS后端文档里面可以不用写尚在完善中，近期要做好的。

File: docs/cn/curvefs-client-design.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- TaskExcutor根据缓存的Copyset信息和集群的拓扑信息，确定rpc需要发送的Ip和端口，之后调用Brpc Client的rpc接口，向MetaServer发送Rpc，去创建Dentry和Inode；
+- 请求首先到达MetaServer的Meta Service层，在MetaService层根据Dentry和Inode所在的Copyset，调用对应的Copyset的接口处理。
+- 在Copyset层，根据Dentry和Inode所在的Partition，调用对应的Partion的接口去处理。
+- 在Partion层，Partition将调用MetaStore接口保存Dentry和Inode
+- MetaStore包含DentryManager和InodeManager两部分，分别将Dentry和Inode存储到DentryStorage和InodeStorage中。
+
+
+### 4.2 Data IO 流
+
+以数据存储到S3为例（数据存储到CURVEBS尚在开发中）：
```
Comment (aspirer) [inline docs/cn/curvefs-client-design.md:68]: 重点讲BS，可以先把设计放上来，实现可以说还在开发。

File: docs/cn/curvefs-client-design.md (hunk 1 above)
Comment (xu-chaojie) [reply docs/cn/curvefs-client-design.md:5]: done

File: docs/cn/curvefs-client-design.md (hunk 2 above)
Comment (xu-chaojie) [reply docs/cn/curvefs-client-design.md:8]: done

File: docs/cn/curvefs-client-design.md (hunk 3 above)
Comment (xu-chaojie) [reply docs/cn/curvefs-client-design.md:2]: done

File: docs/cn/curvefs-client-design.md (hunk 4 above)
Comment (xu-chaojie) [reply docs/cn/curvefs-client-design.md:68]: done

**Review by aspirer — APPROVED** [review]

File: docs/cn/curvefs-client-design.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+CurveFS client包含几个主要模块：
+
+![curvefs_client_structure.drawio.png](./../images/curvefs_client_structure.png)
+
+-   libfuse，对接了其lowlevel fuse api，支持fuse用户态文件系统；
+-   元数据cache，包含fsinfo, inode cache, dentry cache, 实现对元数据的缓存；
+-   meta rpc client， 主要对接元数据集群，实现meta op的发送，超时重试等功能；
+-   S3 client， 通过对接S3接口，将数据存储在S3中；
+-   S3 data cache， 这是S3数据存储的内存缓存层，作为数据缓存，加速S3数据的读写性能；
+-   S3 disk cache，这是S3数据存储的本地持久化缓存，通过磁盘缓存，将对S3数据的读写暂时缓存在本地磁盘上，稍后在异步上传到S3，从而有效降低时延，提供吞吐；
```
Comment (SeanHai) [inline docs/cn/curvefs-client-design.md:43]: 提高吞吐？

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**